src/main/java/com/teragrep/lsh_01/pool/RelpConnectionFactory.java

```diff
@@ -33,18 +33,23 @@ public RelpConnectionFactory(RelpConfig relpConfig) {
     }
 
     @Override
-    public IRelpConnection get() {
+    public IManagedRelpConnection get() {
         RelpConnection relpConnection = new RelpConnection();
 
         RelpConnectionWithConfig relpConnectionWithConfig = new RelpConnectionWithConfig(relpConnection, relpConfig);
+        IManagedRelpConnection managedRelpConnection = new ManagedRelpConnection(relpConnectionWithConfig);
+
+        if (relpConfig.rebindEnabled) {
+            managedRelpConnection = new RebindableRelpConnection(managedRelpConnection, relpConfig.rebindRequestAmount);
+        }
 
         /*
          TODO remove: shouldn't be here, but there is a bug in tearDown, so we initialize connection here
          see https://github.com/teragrep/rlp_01/issues/63 for further info
          */
-        new ManagedRelpConnection(relpConnectionWithConfig).connect();
+        managedRelpConnection.connect();
```

Comment:
if ManagedRelpConnection tracks this internally, there would be no need for external connect calls